Assets/SEE/Game/Avatars/PlayerName.cs

```diff
@@ -87,86 +86,96 @@ public override void OnNetworkDespawn()
+            Log($"{nameof(SendPlayerNameToClientRpc)}(networkObjectId={networkObjectId}, playername={playerName})\n");
+            if (NetworkObjectId == networkObjectId)
+            {
+                SetPlayerName(playerName);
+            }
         }
 
         /// <summary>
-        /// Render the player's name on the TextMeshPro component <see cref="displayNameText"/>
-        /// (and also renames the top-most game object this component is attached to accordingly),
-        /// yet only if the given <paramref name="networkObjectId"/> matches the <see cref="NetworkObjectId"/> of
-        /// this player. Otherwise the player name is for another player (name changes are
-        /// broadcasted to all clients).
+        /// Renders the player's name on the TextMeshPro component <see cref="displayNameText"/>
+        /// (and also renames the top-most game object this component is attached to accordingly).
         /// </summary>
-        /// <param name="networkObjectId"><see cref="NetworkObjectId"/> of the player receiving the player name</param>
-        /// <param name="playername">Player name that should be set</param>
-        private void RenderNetworkPlayerName(ulong networkObjectId, string playername)
+        /// <param name="playerName">Player name that should be set</param>
+        private void SetPlayerName(string playerName)
         {
-            if (NetworkObjectId == networkObjectId)
+            Log($"{nameof(SetPlayerName)}(playername={playerName})\n");
+
+            displayNameText.text = playerName;
+            gameObject.transform.root.name = playerName;
+        }
+
+        /// <summary>
+        /// Renders the player's name on the TextMeshPro component <see cref="displayNameText"/>
+        /// of the child game object of <paramref name="player"/> representing the player's name
+        /// (and also renames the top-most game object this component is attached to accordingly).
+        /// </summary>
+        /// <param name="player">game object representing the player and having a child
+        /// for rendering the player's name</param>
+        /// <param name="nameOfPlayer">Player name that should be set</param>
+        internal static void SetPlayerName(GameObject player, string nameOfPlayer)
+        {
+            Log($"{nameof(SetPlayerName)}(player={player.FullName()}, nameOfPlayer={nameOfPlayer})\n");
+
+            // Name of the immediate child of player representing the player's name.
+            // See the prefabs for players for the structure.
+            const string playerNameGameObjectName = "Playername";
+
+            GameObject child = player.transform.Find(playerNameGameObjectName)?.gameObject;
+
+            if (child == null)
             {
-                displayNameText.text = playername;
-                gameObject.transform.root.name = playername;
+                throw new ($"Player game object {player.FullName()} does not have a child named {playerNameGameObjectName}.");
```

Comment:
I think the space after `new` shouldn't be here:
```suggestion
                throw new($"Player game object {player.FullName()} does not have a child named {playerNameGameObjectName}.");
```